dolphin/api/v1/storages.py

```diff
@@ -175,6 +192,21 @@ def sync(self, req, id):
             LOG.error(e)
             raise exc.HTTPNotFound(explanation=e.msg)
         else:
+            lock = coordination.Lock(storage['id'])
+            with lock:
+                tmp_storage = db.storage_get(ctxt, id)
+                if tmp_storage[constants.DB.DEVICE_SYNC_STATUS] != \
+                        constants.SyncStatus.SYNCED:
+                    msg = 'sync task is running for %s' % tmp_storage['id']
+                    raise exc.HTTPBadRequest(explanation=msg)
+                # Set all bits of sync_status to SYNCING and start sync tasks
+                tmp_storage[constants.DB.DEVICE_SYNC_STATUS] = utils.set_bits(
+                    tmp_storage[constants.DB.DEVICE_SYNC_STATUS],
+                    0,
+                    len(constants.ResourceType) - 1,
+                    constants.SyncStatus.SYNCING)
+                db.storage_update(ctxt, tmp_storage['id'], tmp_storage)
+
```

Comment:
Same code appears at sync_all and sync, please squash it.

Reply:
Done